app/controllers/users_controller.rb

```diff
@@ -0,0 +1,32 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      flash[:danger] = t "error"
+      redirect_to root_path
+    else
+      flash[:success] = t "welcome"
+    end
```

Comment:
Như này thôi nhé, flash welcome kia không phù hợp ngữ cảnh nếu người ta chuyển trang qua lại sau khi đã login rồi
```suggestion
    unless @user
      flash[:danger] = t "error"
      redirect_to root_path
    end
```

Reply:
dạ chị